prov/util/src/util_buf.c

```diff
@@ -49,6 +49,17 @@ int util_buf_grow(struct util_buf_pool *pool)
 	ssize_t hp_size;
 	struct util_buf_footer *buf_ftr;
 
+	if ((pool->num_allocated > 0) && pool->attr.unavail_hndlr) {
```

Comment:
Did you mean `pool->attr.max_cnt && pool->num_allocated >= pool->attr.max_cnt && pool->attr.unavail_hndlr` like the check below?